packages/lib-panoptes-js/src/experimental-auth.js

```diff
@@ -0,0 +1,238 @@
+    // fetching the user resource, which is, uh, kinda extra work, but keeps the
+    // calls standardised I guess?
+
+    // Step 3: get the bearer token.
+
+    const request3 = new Request(`https://panoptes-staging.zooniverse.org/oauth/token`, {
+      body: JSON.stringify({
+        client_id: '535759b966935c297be11913acee7a9ca17c025f9f15520e7504728e71110a27',
+        grant_type: 'password',
+      }),
+      credentials: 'include',
+      method: 'POST',
+      headers: PANOPTES_HEADERS,
+    })
+    const response3 = await fetch(request3)
+    console.log('+++ Step 3: ', response3)
+
+    // Extract data and check for errors.
+    if (!response3.ok) {
+      const jsonData3 = await response3.json()
+      const error = jsonData3?.error || 'No idea what went wrong; no specific error message detected.'
+      throw new Error(`Error from API. ${error}`)
+    }
+    const jsonData3 = await response3.json()
+    const bearerToken = jsonData3?.access_token  // The bearer token is short-lived
+    const refreshToken = jsonData3?.refresh_token  // The refresh token is used to get new bearer tokens.
+    const bearerTokenExpiry = Date.now() + (jsonData3?.expires_in * 1000)  // Use Date.now() instead of response.created_at, because it keeps future "has expired?" comparisons consistent to the client's clock instead of the server's clock.
+    if (!bearerToken || !refreshToken) {
+      throw new Error('Impossible API response. access_token and/or refresh_token unavailable.')
+    } else if (jsonData3?.token_type !== 'Bearer') {
+      throw new Error('Impossible API response. Token wasn\'t of type "Bearer".')
+    } else if (isNaN(bearerTokenExpiry)) {
+      throw new Error('Impossible API response. Token expiry can\'t be calculated.')
+    } else if (bearerTokenExpiry <= Date.now()) {
+      throw new Error('Impossible API response. Token has already expired for some reason.')
+    }
+
+    console.log('+++ signIn() Results: ',
+      '\n\n  userData:', userData,
+      '\n\n  bearerToken', bearerToken,
+      '\n\n  bearerTokenExpiry', new Date(bearerTokenExpiry),
+      '\n\n  refreshToken', refreshToken,
+    )
+    _broadcastEvent('change', userData)
+
+  } catch (err) {
+    console.error('+++ ERROR: signIn() ', err)
+  }
+
+  /*
+  Original PJC code:
+
+  const user = await this.checkCurrent();
+  if (user) {
+    await this.signOut();
+    return this.signIn(credentials);
+  } else {
+    console.log('Signing in', credentials.login);
+    const token = await getCSRFToken(config.host)
+    const data = {
+      authenticity_token: token,
+      user: {
+        login: credentials.login,
+        password: credentials.password,
+        remember_me: true,
+      },
+    };
+
+    const signInRequest = this._makeSignInRequest(data);
+    this._currentUserPromise = signInRequest.catch(() => null);
+    await this._currentUserPromise;
+    this.emit('change', this._currentUserPromise);
+
+    return signInRequest;
+  }
+  */
+}
+
+export {
```

Comment:
You’ll need to change the package type to `module` if you want to use `export` in Node. PJC only runs in browsers but `lib-panoptes-js` is used in Node and in browsers.